keps/sig-architecture/3935-oldest-node-newest-control-plane/README.md

```diff
@@ -0,0 +1,535 @@
+    * Supporting n-3 nodes would have delayed removal of the in-tree volume plugin to v1.28.
+* Supported skew between node components
+  * ⚠️ sig-network: `IPTablesOwnershipCleanup` changes for [KEP-3178](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/3178-iptables-cleanup#version-skew-strategy)
+    made v1.27+ `kubelet` instances rely on `kube-proxy` being >= v1.25. Clusters that deploy older `kube-proxy` instances alongside newer `kubelet` instances could not support n-3 skew between them.
+    * Supporting n-3 nodes would have delayed promotion of `IPTablesOwnershipCleanup` from v1.27 to v1.28.
+* Removal of deprecated REST APIs used by node components ([source](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-27))
+  * ✅ None
+
+### Planned control plane changes
+
+Are planned v1.28 control plane changes as compatible with v1.25 nodes as n-2 nodes?
+* Feature enablement in control plane components that will wait for node support
+  * ...
+* Removal of deprecated behavior in control plane components
+  * ✅ sig-storage: planned removal of in-tree volume plugins for `CSIMigrationGCE` feature, which is guaranteed on in v1.25 nodes ([#111301](https://github.com/kubernetes/kubernetes/pull/111301).
+  * ...
+* Removal of deprecated REST APIs used by node components ([source](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-28))
+  * ✅ None
+
+Are planned v1.29 control plane changes as compatible with v1.26 nodes as n-2 nodes?
+* Feature enablement in control plane components that will wait for node support
+  * ...
+* Removal of deprecated behavior in control plane components
+  * ⚠️ sig-storage: planned removal of in-tree volume plugins for `CSIMigrationvSphere` and `CSIMigrationAzureFile` features, which are guaranteed on in v1.27 nodes ([#116610](https://github.com/kubernetes/kubernetes/pull/116610) and [#114953](https://github.com/kubernetes/kubernetes/pull/114953)).
+    * Supporting n-3 nodes would delay removal of the in-tree volume plugins to v1.30.
+  * ...
+* Removal of deprecated REST APIs used by node components ([source](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-29))
+  * ✅ None
+
+Are planned v1.30 control plane changes as compatible with v1.27 nodes as n-2 nodes?
+* Feature enablement in control plane components that will wait for node support
+  * ...
+* Removal of deprecated behavior in control plane components
+  * ...
+* Removal of deprecated REST APIs used by node components ([source](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-30))
+  * ✅ None
+
+### Impact summary
+
+The calculated impact over the past two years if Kubernetes supported n-3 node skew:
+* the sig-auth `BoundServiceAccountTokenVolume` feature GA would have moved from v1.22 to v1.23 (remaining in beta and enabled by default in v1.22).
+* the sig-auth Pod Security update for the `IdentifyPodOS` feature would have moved from v1.25 to v1.26.
```

Comment:
IIRC we debated holding the Pod Security updates for `IdentifyPodOS` until v1.26 and the decision add the functionality in v1.25 and relax validation for Windows nodes was intentional

Reply:
Yeah, because 1.23 kubelets were the oldest supported and were the first to safely refuse to run a mismatched podOS. If n-3 skew was in place then, that would have delayed relaxing Pod Security to 1.26